lib/generators/engine_cart/install_generator.rb

```diff
@@ -52,8 +52,7 @@ def add_gemfile_include
   # the below comes from engine_cart, a gem used to test this Rails engine gem in the context of a Rails app
   file = File.expand_path("Gemfile", ENV['ENGINE_CART_DESTINATION'] || ENV['RAILS_ROOT'] || File.expand_path("../spec/internal", __FILE__))
   if File.exist?(file)
-    puts "Loading \#{file} ..." if $DEBUG # `ruby -d` or `bundle -v`
-    instance_eval File.read(file)
+    eval_gemfile file
   else
     # we get here when we haven't yet generated the testing app via engine_cart
     gem 'rails', ENV['RAILS_VERSION'] if ENV['RAILS_VERSION']
```

Comment:
Oh good point, you're right, that's probably be a roadblock, going to cause even more problems like the ones you're trying to solve in the first place. 

Okay, as a workaround, the README should probably have copyable text, so when you're upgrading you know where to get/check the current text that should be in Gemfile. But then it has to be kept in sync with code, that I know we're not always good at keeping docs in sync with code. Or maybe just a link to the source in github to copy and paste from. @ndushay , in case you want to try to do something with this in your README updates (which are so so welcome).